### DISCUSSION TITLE
API is very confusing by listing Stars count for Watchers count on all repos

### DISCUSSION THREAD

**Discussion by vtempest** [community/community#24795] [Apps, API and Webhooks]
<pre><code>https://api.github.com/search/repositories?q=vscode
stargazers_count: 70274,
forks_count: 9362,
forks: 9362
watchers_count: 70274, &lt;--(LISTED)
watchers: 70274,

https://github.com/Microsoft/vscode
Stars 70,274
Forks 9,362
Watchers 2,667 &lt;--- (REAL)
</code></pre>

**Comment by rsese** [comment, marked as answer]
<p>Sorry for the confusion here - this part of the API changed when we moved “watching” to mean “starring” and is explained in this post from the API blog:</p>
<p><a href="http://developer.github.com/changes/2012-9-5-watcher-api/" rel="nofollow noopener">http://developer.github.com/changes/2012-9-5-watcher-api/</a></p>
<p><code>watchers</code> corresponds to the number of users that have starred a repository, while <code>subscribers_count</code> corresponds to the number of watchers. For example, looking at the Atom repository:</p>
<p><a href="https://api.github.com/repos/atom/atom" rel="nofollow noopener">https://api.github.com/repos/atom/atom</a></p>
<p>Right now it has:</p>
<p>“subscribers_count”: 2468<br>
“stargazers_count”: 48009</p>
<p>This is because there are 2468 watchers (= subscribers) and 48009 stargazers.</p>
<p>I hope that clears things up but if you have more questions, please let us know!</p>

**Comment by jinyeqing** [comment]
<p>repo_list = <span class="chcklst-box fa fa-square-o fa-fw"></span><br>
for (var j = 0; j &lt; Pages.length; j++) {<br>
if (!Pages[j][‘fork’]) {//排除fork别人的库<br>
repo_list.push({<br>
name: Pages[j][‘name’],<br>
stargazers_count: Pages[j][‘stargazers_count’],<br>
forks_count: Pages[j][‘forks_count’],<br>
html_url: Pages[j][‘html_url’],<br>
open_issues_count: Pages[j][‘open_issues_count’],<br>
subscribers_count: Pages[j][‘subscribers_count’],<br>
language: Pages[j][‘language’]</p>
<p>});<br>
}<br>
}</p>
<p>why i use this，but show like the photo as follows：<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/92c0d3f4df82806c2eceed8ecf55b922afca9f43.png" data-download-href="/uploads/short-url/kWeQfeS8Obzp5znkbljP2rDS8w3.png?dl=1" title="WechatIMG504.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/14799589/181086900-e8ff9f37-32d2-41ed-9a63-e9c3c76dce92.png" alt="WechatIMG504.png" data-base62-sha1="kWeQfeS8Obzp5znkbljP2rDS8w3" width="690" height="301" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/92c0d3f4df82806c2eceed8ecf55b922afca9f43_2_690x301.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/92c0d3f4df82806c2eceed8ecf55b922afca9f43_2_1035x451.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/9/92c0d3f4df82806c2eceed8ecf55b922afca9f43_2_1380x602.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/92c0d3f4df82806c2eceed8ecf55b922afca9f43_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">WechatIMG504.png</span><span class="informations">2550×1114 528 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by rsese** [comment]
<p>Hmm, not sure what the issue just from the code screenshot but are you able to reproduce the problem with <code>subscribers_count</code> with a <code>curl</code> request?  For example, I see <a href="https://github.com/AgoraIO/Electron-SDK" rel="nofollow noopener">https://github.com/AgoraIO/Electron-SDK</a> in your list of repositories but if I get that repository [1] with <code>curl</code>:</p>
<pre><code>$ curl https://api.github.com/repos/AgoraIO/Electron-SDK | grep subscribers_count

  "subscribers_count": 34
</code></pre>
<p>We see <code>subscribers_count</code> in the response.  If you can’t reproduce with <code>curl</code>, my guess is that there’s an issue with the code.</p>
<p>1: <a href="https://developer.github.com/v3/repos/#get" rel="nofollow noopener">https://developer.github.com/v3/repos/#get</a></p>
